chore(core): enable deletion of more than 10 ssm parameters at a time

twoquarterrican opened the pull request:
Cross-region-ssm-writer-handler fails when trying to delete more than 10 ssm parameters

This feature is used when exporting resources from a stack in one region to a stack in another region.
It uses the `crossRegionReferences: true` flag in the stack.

The reason for this change is the following error I received on a stack update (slightly modified to hide some names)

```
❌ Deployment failed: Error: The stack named NomCoreStack failed to deploy: UPDATE_ROLLBACK_FAILED (The following resource(s) failed to update: [ExportsWriteruseast2828FA26B86FBEFA7]. ): Received response status [FAILED] from custom resource. Message returned: ValidationException: 1 validation error detected: Value '[/cdk/exports/NomServicesStack/NomCoreStackxohpRefUserPoolIdentityPoolA58D72D6C9223D28, /cdk/exports/NomIntegrationTestsStack/NomCoreStackxohpRefUserPoolUserPoolClient40176907C34D1493, /cdk/exports/NomLeafStack/NomCoreStackxohpRefUserPoolIdentityPoolA58D72D6C9223D28, /cdk/exports/NomLeafStack/NomCoreStackxohpFnGetAttUserPool6D0DFADBArnD191ECDE, /cdk/exports/NomCoreStack/NomCoreStackxohpRefUserPoolUserPoolClient40176907C34D1493, /cdk/exports/NomIntegrationTestsStack/NomCoreStackxohpFnGetAttUserPool6D0DFADBArnD191ECDE, /cdk/exports/NomIntegrationTestsStack/NomCoreStackxohpRefUserPoolIdentityPoolA58D72D6C9223D28, /cdk/exports/NomServicesStack/NomCoreStackxohpFnGetAttUserPool6D0DFADBArnD191ECDE, /cdk/exports/NomLeafStack/NomCoreStackxohpRefUserPoolUserPoolClient40176907C34D1493, /cdk/exports/NomServicesStack/NomCoreStackxohpRefUserPoolUserPoolClient40176907C34D1493, /cdk/exports/NomLeafStack/NomCoreStackxohpRefUserPoolUserPoolDomain9F01E991C941B942]' at 'names' failed to satisfy constraint: Member must have length less than or equal to 10
```

Design for the fix is simple: Break the list of names to delete into chunks of at most 10. Invoke delete as many times as needed to delete all. Throttling is not handled here. This may come into play when there are 50 or more parameters to delete, which may not be a concern worth addressing at this point.


Closes No issue created for this

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

aws-cdk-automation reviewed (dismissed):
The pull request linter has failed. See the aws-cdk-automation comment below for failure reasons. If you believe this pull request should receive an exemption, please comment and provide a justification.


A comment requesting an exemption should contain the text `Exemption Request`. Additionally, if clarification is needed add `Clarification Request` to a comment.

aws-cdk-automation commented:
The pull request linter fails with the following errors:

	❌ Fixes must contain a change to an integration test file and the resulting snapshot.

<b>PRs must pass status checks before we can provide a meaningful review.</b>

If you would like to request an exemption from the status checks or clarification on feedback, please leave a comment on this PR containing `Exemption Request` and/or `Clarification Request`.

aws-cdk-automation commented:
The pull request linter fails with the following errors:

	❌ Fixes must contain a change to an integration test file and the resulting snapshot.

<b>PRs must pass status checks before we can provide a meaningful review.</b>

If you would like to request an exemption from the status checks or clarification on feedback, please leave a comment on this PR containing `Exemption Request` and/or `Clarification Request`.

aws-cdk-automation commented:
The pull request linter fails with the following errors:

	❌ Fixes must contain a change to an integration test file and the resulting snapshot.

<b>PRs must pass status checks before we can provide a meaningful review.</b>

If you would like to request an exemption from the status checks or clarification on feedback, please leave a comment on this PR containing `Exemption Request` and/or `Clarification Request`.

aws-cdk-automation commented:
The pull request linter fails with the following errors:

	❌ Fixes must contain a change to an integration test file and the resulting snapshot.

<b>PRs must pass status checks before we can provide a meaningful review.</b>

If you would like to request an exemption from the status checks or clarification on feedback, please leave a comment on this PR containing `Exemption Request` and/or `Clarification Request`.

aws-cdk-automation commented:
The pull request linter fails with the following errors:

	❌ Fixes must contain a change to an integration test file and the resulting snapshot.

<b>PRs must pass status checks before we can provide a meaningful review.</b>

If you would like to request an exemption from the status checks or clarification on feedback, please leave a comment on this PR containing `Exemption Request` and/or `Clarification Request`.

twoquarterrican commented:
Clarification Request

My change fixes some issues only seen during a stack update or delete operation. I cannot tell how to integration test these operations. Currently the pull request linter is asking me to make a change to the integration test file. I cannot see any meaningful change I can make to test the behavior that I modified. I have already updated the resulting snapshot, so all that remains is to either integration test an update or delete operation, or accept the pull request with the unit tests as sufficient verification of the update / delete operation enhancements

aws-cdk-automation commented:
The pull request linter fails with the following errors:

	❌ Fixes must contain a change to an integration test file and the resulting snapshot.

<b>PRs must pass status checks before we can provide a meaningful review.</b>

If you would like to request an exemption from the status checks or clarification on feedback, please leave a comment on this PR containing `Exemption Request` and/or `Clarification Request`.

aws-cdk-automation commented:
The pull request linter fails with the following errors:

	❌ Fixes must contain a change to an integration test file and the resulting snapshot.

<b>PRs must pass status checks before we can provide a meaningful review.</b>

If you would like to request an exemption from the status checks or clarification on feedback, please leave a comment on this PR containing `Exemption Request` and/or `Clarification Request`.

TheRealAmazonKendra commented:
I also think this is probably more of a chore than a fix because the API call is behaving as intended but we can improve the experience here by taking your change. So, I've updated the title to be a chore and to add a little clarity.

TheRealAmazonKendra reviewed (approved):
Thank you for your contribution!
